game/world/managers/objects/script/ScriptHandler.py

```diff
@@ -141,6 +140,8 @@ def handle_script_command_talk(command):
 
         speaker = command.target if command.target and command.target.get_type_id() == ObjectTypeIds.ID_UNIT \
             else command.source
+        if command.source.quest_target is not None:
+            command.target = command.source.quest_target
```

Comment:
Hmm are you sure this is needed? I think normal targeting should cover all cases, for example `Follow me, $N. I'll take you to the Defias hideout. But you better protect me or I am as good as dead.` is sent when accepting the quest.

Reply:
For *that* phrase, yes. However, this does no longer work at the end of the escort so I had to implement this.

Reply:
Seems like for escorts, the script should be an "scripted map event" and the player on these cases would be obtained through `TARGET_T_MAP_EVENT_TARGET`.

Reply:
We'll probably need to implement some kind of MapScriptManager or similar in order to handle all of this, but I guess I can merge this for now and we can slowly work on it. :+1: